internal/controller/util.go

```diff
@@ -7,36 +7,15 @@ package controller
 import (
 	"fmt"
 
-	"golang.org/x/exp/slices"
 	appsv1 "k8s.io/api/apps/v1"
 	v1 "k8s.io/api/core/v1"
 
 	temporaliov1alpha1 "github.com/DataDog/temporal-worker-controller/api/v1alpha1"
 	"github.com/DataDog/temporal-worker-controller/internal/controller/k8s.io/utils"
 )
 
-func findHighestPriorityStatus(statuses []temporaliov1alpha1.ReachabilityStatus) temporaliov1alpha1.ReachabilityStatus {
-	if len(statuses) == 0 {
-		return ""
-	}
-	slices.SortFunc(statuses, func(a, b temporaliov1alpha1.ReachabilityStatus) int {
-		return getStatusPriority(a) - getStatusPriority(b)
-	})
-	return statuses[len(statuses)-1]
-}
-
-func getStatusPriority(s temporaliov1alpha1.ReachabilityStatus) int {
-	switch s {
-	case temporaliov1alpha1.ReachabilityStatusReachable:
-		return 4
-	case temporaliov1alpha1.ReachabilityStatusClosedOnly:
-		return 3
-	case temporaliov1alpha1.ReachabilityStatusUnreachable:
-		return 2
-	case temporaliov1alpha1.ReachabilityStatusNotRegistered:
-		return 1
-	}
-	return 0
+func computeVersionID(spec *temporaliov1alpha1.TemporalWorkerSpec) string {
+	return spec.WorkerOptions.DeploymentName + "." + computeBuildID(spec)
 }
 
 func computeBuildID(spec *temporaliov1alpha1.TemporalWorkerSpec) string {
```

Comment:
follow-up PR: include user-passed build id / version id in this as a human-readable portion